### DISCUSSION TITLE
fiber tracts clustered around midbrain

### DISCUSSION THREAD

**Discussion by iliana-karipidis** [yeatmanlab/pyAFQ#928] [General]
I am setting up a new DWI protocol and analyzing some pilot data (Philips scanner, 32 dirs, b=1,000 & 2,000, isotropic 2mm). Preprocessing was performed using qsiprep and qsirecon, output was reviewed and looks ok. The pyAFQ output looks incorrect, with all fiber tracts clustered around the midbrain. What went wrong?
<img width="612" alt="pilot_pyAFQ_output" src="https://user-images.githubusercontent.com/67779741/205512129-1a249df5-195c-46be-a448-077b9a3be610.png">

This issue seems to be specific to this pilot data, processing single-shell data of another sequence (Philips scanner, 32 dirs, b=1,000, isotropic 2mm) worked well:
<img width="612" alt="original_pyAFQ_output" src="https://user-images.githubusercontent.com/67779741/205512787-16e1aeb6-2b4c-408c-abaa-a4b14e6e4fdb.png">

**Comment by arokem** [comment]
Hi @iliana-karipidis : 

If I had to guess something went wrong in the mapping stage. I would look at the file that has the suffix: "dwi_b0_in_MNI.nii.gz". I imagine it looks very wrong. If that is the case, there are two options here: 

1. You used the mapping provided by qsiprep. That is, your pyAFQ config file had a section that looks like this: 

```
[MAPPING]
mapping_definition = "ItkMap(warp_suffix='xfm', warp_filters={'from': 'MNI152NLin2009cAsym', 'to': 'T1w', 'scope': 'qsiprep'})"
```
If you did that, I would suggest trying to run your pipeline without reference to the qsiprep mapping. I.e., take out this section from your config.

2. You used the mapping generated by pyAFQ. In this case, you have two options: 
    a. Add a section like the above to your config, to use the qsiprep mapping. This should also make the processing a little bit faster. 
    b. Take a look at the file that ends with a "CSD_APM.nii.gz" suffix. The data there should look roughly like a T1-weighted image. This is also a place to look and see whether the brain mask looks reasonable. That is, does the brain mask cover all of the brain, and only the brain.

**Reply by iliana-karipidis** [reply]
Thanks @arokem !
I used the default and thus the mapping generated by pyAFQ. Your guess is correct, something went wrong with the mapping stage:
<img width="922" alt="pyafq_mapping" src="https://user-images.githubusercontent.com/67779741/205716480-65adc96e-d5e0-4a01-b19c-caaeb7a61e67.png">


I followed option 2 and added the mapping section to the config file. When using the qsiprep mapping the "dwi_b0_in_MNI.nii.gz" file looks okay. The "CSD_APM.nii.gz" file still looks wrong and the resulting bundles are also not nice:
<img width="945" alt="qsiprep_mapping" src="https://user-images.githubusercontent.com/67779741/205716512-e39cb21f-d28a-4251-a5f7-68258fc1f21e.png">

Do you have any further suggestions what might be causing this?

**Comment by arokem** [comment]
Also, just to be sure: the sequences used are identical apart from the addition of a second shell?

**Reply by iliana-karipidis** [reply]
Yes, that's correct - I just added a second shell to the sequence. Here's how the single-shell data looks like:
<img width="936" alt="singleshell_pyafq_mapping" src="https://user-images.githubusercontent.com/67779741/205717877-44532f13-465d-449b-887e-687da3320645.png">

**Comment by arokem** [comment]
I'll start by saying that I am not sure what is going on here, and I need to think about this more (maybe others have more ideas).

What model are you using for tractography in the two-shell data? I don't think that mrtrix's standard CSD would work here because I don't think that it works out of the box on multi-shell data. Or maybe you are using the multi-shell multi-tissue tractography here? In addition, it looks like CSD fit to the two shells (which DIPY's implementation does do) doesn't work very well -- the APM is computed from the CSD coefficients. I don't think that I've ever seen an APM map with these very high-intensity spots.

**Reply by iliana-karipidis** [reply]
I used the mrtrix_multishell_msmt_ACT-hsvs workflow for reconstruction: https://qsiprep.readthedocs.io/en/latest/reconstruction.html#mrtrix-multishell-msmt-act-hsvs

**Comment by arokem** [comment]
On a bit more though, could you share your config file? I figure that can answer a few different questions that I have.

**Reply by iliana-karipidis** [reply]
Here's the config file I used:

```toml
[BIDS_PARAMS]

# The path to preprocessed diffusion data organized in a BIDS
# dataset. This should contain a BIDS derivative dataset with
# preprocessed dwi/bvals/bvecs.
bids_path = 'human/data_32_dir_2shells'

# The name of the pipeline used to preprocess the DWI data.
# Default: "all".
dmriprep = 'qsiprep'


[MAPPING]
mapping_definition = "ItkMap(warp_suffix='xfm', warp_filters={'from': 'MNI152NLin2009cAsym', 'to': 'T1w', 'scope': 'qsiprep'})"
```
